libbpfgo/libbpfgo.go

```diff
@@ -598,14 +623,84 @@ func doAttachKprobeLegacy(prog *BPFProg, kp string, isKretprobe bool) (*BPFLink,
 var eventChannels = make(map[uintptr]chan []byte)
 var lostChannels = make(map[uintptr]chan uint64)
 
+func (m *Module) InitRingBuf(mapName string, eventsChan chan []byte) (*RingBuffer, error) {
+	bpfMap, err := m.GetMap(mapName)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := uintptr(bpfMap.fd)
+	if eventsChan == nil {
+		return nil, fmt.Errorf("events channel can not be nil")
+	}
+	eventChannels[ctx] = eventsChan
+
+	rb := C.init_ring_buf(bpfMap.fd)
+	if rb == nil {
+		return nil, fmt.Errorf("")
+	}
+
+	ringBuf := &RingBuffer{
+		rb:     rb,
+		bpfMap: bpfMap,
+		stop:   make(chan bool),
+	}
+	m.ringBufs = append(m.ringBufs, ringBuf)
+	return ringBuf, nil
+}
+
+func (rb *RingBuffer) Start() {
+	go rb.poll()
+}
+
+func (rb *RingBuffer) Stop() {
+	rb.stop <- true
+}
+
+func (rb *RingBuffer) Close() {
+	if rb.closed {
+		return
+	}
+	for {
+		if rb.stopped {
+			break
+		}
+	}
```

Comment:
not sure I understand: if Close was called before Stop, you wait for Stop to have happened? isn't it better to call Stop instead?

Reply:
Yes good point, need to add a check in Stop() to see if it's already stopped because of the nature of channels but this makes more sense

Reply:
This throws off the fact that there's only one consumer of `rb.stop` so can cause a block. Now that I think of it, do we even need a `Stop` and a `Close`? Who is going to stop a ringbuffer and then restart it later? I'd be fine with just removing `Stop` and having the write to `rb.stop` in `Close()`. 

cc @yanivagman

Reply:
I'll let you and Yaniv decide if you want to support start after stop (make sure to document it) but certainly we don't want to support start after close, so I think either way close should just call stop, no?